### PR TITLE
Fix POST queries to honour the abort signal

### LINKED ISSUE #3710: Query cancellation not working
Evident when clicking around on a Druid-backed dashboard and looking at the browser's network tab. It's also evident when looking at a large volume DuckDB-backed dashboard.

[Reported in Slack](https://rilldata.slack.com/archives/C0415QK8DK6/p1702667416409829)

### PR DESCRIPTION
**Pull request by AdityaHegde**
closes #3710

## Checklist
- [x] Manual verification
- [ ] Unit test coverage
- [ ] E2E test coverage
- [x] Needs manual QA?

#### Issue addressed: 
The queries in dashboard are not being cancelled when changing the filters and other settings. This is apparent while using druid or a large project.

#### Details:
We actually call cancel queries. But the orval generated code is not passing on the abort signal to our client. So cancellations are not honoured for these. The root cause is that these queries are POST requests overridden to be queries instead of mutations. This seems to have an edge case in orval of not adding signal properly.

### CODE REVIEW

File: tsconfig.json
Comment (ericpgreen2) [inline tsconfig.json:1]: Do we have to remove the comments in this file?

File: web-common/src/runtime-client/add-signal-argument.ts
Comment (ericpgreen2) [inline web-common/src/runtime-client/add-signal-argument.ts:1]: I really like how this is encapsulated into this one file. To help our future selves – Could you add a comment at the top of this file to 1) explain the motivation, and 2) describes the input / output of the transformation?

**Review by ericpgreen2 — COMMENTED** [review]
Overall, I like the encapsulation of the transformation. Some comments inline, and also:
- How do you feel about how resilient it is to time?
- Did you check with Orval community to see if anyone had run into this issue before? I think the maintainer is quite responsive, so creating a quick issue could be good.

File: tsconfig.json
Comment (AdityaHegde) [reply tsconfig.json:1]: Nope. This was a temporary change.

**Comment by AdityaHegde** [comment]
It is not guaranteed to be stable with newer versions. I have fixed the version of orval for now.

Ideally there should be a fix from orval. Created a discussion for now: https://github.com/anymaniax/orval/discussions/1151

**Comment by ericpgreen2** [comment]
In trying to figure out if query cancellation is working, I found myself in this state after navigating quickly away from a loading dashboard, then returning to it:

image

**Pull request merged**